### PR TITLE
restructure repo and add poetry

### PR DESCRIPTION
**Pull request by bardkw**
The purpose of this PR is to restructure the repository and add poetry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+license = "BSD-3-Clause"
+
+[tool.poetry.dependencies]
+python = "^3.6"
+setuptools = ">=53"
+numpy = "^1.19.5"
+pillow = "^8.2.0"
+smqtk-core = "^0.17.0"
+smqtk-dataprovider = "^0.15.0"
+smqtk-classifier = "^0.15.0"
```
Comment (Purg) [inline pyproject.toml]: Lets use to `>=` qualifier for our `0.*` minor versions (`^0.17.0` specifically bounds within the `0.17.*` version range), and update to versions 0.18.0, 0.16.0 and 0.16.0 respectively. Indexing package version update pending my finishing of Kitware/SMQTK-Indexing#8

File: .mypy.ini [hunk 2]
```diff
@@ -2,60 +2,34 @@
...
+;[mypy-tests.*]
 
 ###############################################################################
 # TPL-module options
 
 [mypy-numpy.*]
 ignore_missing_imports = True
 
-[mypy-PIL.*]
+[mypy-pkg_resources.extern]
```
Comment (Purg) [inline .mypy.ini]: Remember to remove with the joined-`setuptools` dep removal

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+[tool.poetry]
+name = "smqtk_relevancy"
+version = "0.15.0"
+description = "SMQTK Relevancy"
+authors = ["Kitware, Inc. <smqtk-developers@kitware.com>"]
+license = "BSD-3-Clause"
+
+[tool.poetry.dependencies]
+python = "^3.6"
+setuptools = ">=53"
```
Comment (Purg) [inline pyproject.toml]: Remove setuptools dep

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+sphinx-rtd-theme = "^0.5.1"
+sphinx-argparse = "^0.2.5"
+sphinx-prompt = "^1.4.0"
+livereload = "^2.6.3"
+# Testing
+coverage = "^5.5"
+mock = "^4.0.3"
+pytest = "^6.2.2"
+pytest-cov = "^2.11.1"
+
```
Comment (Purg) [inline pyproject.toml:42]: With the other packages I've started adding a python 3.6 compatible `ipython` and `jedi` verison for common development purposes.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+###############################################################################
+[tool.pytest.ini_options]
+addopts = [
+    "-lv",                  # Show local in trace-backs.
+    "--doctest-modules",    # Increased verbosity.
+    "--tb=long",            # Trace-back print mode.
```
Comment (Purg) [inline pyproject.toml:48]: As part of the other repo branches i've also been adding `--cov=<PACKAGE_NAME>` and `--cov-report=term` here to output that info. I've been finding that I'm often adding those manually, so thought why not reduce the typing.

File: smqtk_relevancy/__init__.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import pkg_resources
```
Comment (Purg) [inline smqtk_relevancy/__init__.py]: Continuing with the theme, lets remove this import and the `__version__` attr below.

File: smqtk_relevancy/utils/distance_kernel.py [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+        for i in range(m1.shape[0]):
+            k[i, :] = dist_func(m1[i], m2)
+    else:
+        for i in range(m1.shape[0]):
+            for j in range(m2.shape[0]):
+                k[i, j] = dist_func(m1[i], m2[j])
+    return k
+
+
+class DistanceKernel (object):
```
Comment (Purg) [inline smqtk_relevancy/utils/distance_kernel.py]: This class can likely be removed. I'd say double check that it's not being used, but I don't think it is.

File: smqtk_relevancy/utils/distance_kernel.py [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+                mat[i, i] = dist_func(m[i], m[i])
+                # cols to the left of diagonal index for this row
+                for j in range(i):
+                    mat[i, j] = mat[j, i] = dist_func(m[i], m[j])
+                pr.increment_report()
+    pr.report()
+    return mat
+
+
+def compute_distance_matrix(m1, m2, dist_func, row_wise=False):
```
Comment (Purg) [inline smqtk_relevancy/utils/distance_kernel.py:12]: Does this function get used anywhere?

File: smqtk_relevancy/utils/read_write_lock.py [hunk 9]
```diff
@@ -0,0 +1,486 @@
+"""
+LICENCE
+-------
+Copyright 2013-2020 by Kitware, Inc. All Rights Reserved. Please refer to
+LICENSE.TXT for licensing information, or contact General Counsel,
+Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.
+"""
+
+from collections import deque
```
Comment (Purg) [inline smqtk_relevancy/utils/read_write_lock.py]: I think this is only used in that distance kernel class that can be removed, so this file may also be removed I think.

**Comment by bardkw** [comment]
setuptools dep removed and resolved other issues. should be good to go now

File: pyproject.toml [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+description = "SMQTK Relevancy"
+authors = ["Kitware, Inc. <smqtk-developers@kitware.com>"]
+license = "BSD-3-Clause"
+
+[tool.poetry.dependencies]
+python = "^3.6"
+numpy = "^1.19.5"
+pillow = "^8.2.0"
+smqtk-core = ">=0.18.0"
+smqtk-dataprovider = ">=0.15.0"
```
Comment (Purg) [inline pyproject.toml]: Maybe for update later, but these should all be at >=0.16.0 now.

**Review by Purg — APPROVED** [review]
Just a not-immediately-important version update comment, but otherwise LGTM

File: pyproject.toml (hunk 10 above)
Comment (bardkw) [reply pyproject.toml]: updated

**Pull request merged**